eventbrite-consents/src/main/scala/com/gu/identity/eventbriteconsents/clients/IdentityClient.scala

```diff
@@ -0,0 +1,21 @@
+package com.gu.identity.eventbriteconsents.clients
+
+import io.circe.generic.auto._
+import io.circe.syntax._
+import scalaj.http.Http
+
+case class IdapiConsentUpdate(email: String, `set-consents`: Vector[String])
+
+class IdentityClient(idapiUrl: String, idapiAccessToken: String) {
+  def updateEventConsent(emailAddress: String): Unit = {
+    val result = Http(s"$idapiUrl/consent-email")
+      .headers(
+        "X-GU-ID-Client-Access-Token" -> s"Bearer $idapiAccessToken",
+        "Content-type" -> "application/json"
+      )
+      .postData(IdapiConsentUpdate(emailAddress, Vector("events")).asJson.noSpaces)
+      .asString
+
+    if (result.code != 200) throw new RuntimeException(s"Unexpected response from idapi ${result.code} ${result.body}")
+  }
+}
```

Comment:
It's unclear to me how we are handling errors on this end. Id Identity is down, will this retry to update the email? Might we not loose the occasional consent here as each one is  separate request to Identit that has to succeed?

Reply:
I'm concerned about implementing a retry mechanism since sending emails is not idempotent - retrying the action might result in someone being spammed in the scenario where identity successfully sends the email but some error (e.g. network failure) prevents the successful result being returned to the lambda. 

The choice is:

- users always receive at least one or more emails when they consent (where they are potentially spammed by identity)
- a user receives no more than one email (the user might not receive an email on failure)

There are pager duty alerts for lambda failures. I will add more logging to make it clear which emails the service received successful responses for, and which ones were never triggered.

Reply:
ok. This makes sense as it's just the email getting triggered not the actual consent being added to Identity. I guess we need to accept that we will loose the occasional signup but I imagine this will be rare. If we set up an alert for a (highish) threshold of failures we will a least know if all the emails in a sync failed because IDAPI was having issues at that moment.